docs/index.md

```diff
@@ -0,0 +1,89 @@
+# Columbo
+
+[![CI pipeline status](https://github.com/wayfair-incubator/columbo/workflows/CI/badge.svg?branch=main)][ci]
+[![codecov](https://codecov.io/gh/wayfair-incubator/columbo/branch/main/graph/badge.svg)][codecov]
+TODO: PyPi Release Badge
+
+`columbo` provides a way to specify a dynamic set of questions to ask a user and get their answers.
+
+## Example
+
+### User Prompts
+
+The primary use of `columbo` is to define a sequence of interaction that are used to prompt a user to provide answers
+using a terminal. Below is a sample which shows how this can be used.
+
+```python
+import columbo
+
+interactions = [
+    columbo.Echo("Welcome to the Columbo example"),
+    columbo.Acknowledge(
+        "Press enter to start"
+    ),
+    columbo.BasicQuestion(
+        "user",
+        "What is your name?",
+        default="Patrick",
+    ),
+    columbo.BasicQuestion(
+        "user_email",
+        lambda answers: f"""What email address should be used to contact {answers["user"]}?""",
```

Comment:
I know this is a basic example, but might be worth showing assigning the `default` arg to a callable that isnt defined inline. this works as a basic example, but it would be nice to show off the function signature of a `default` resolver function

Reply:
I'm trying to balance having a short example and showing features. It will be fully covered in the `Usage Guide`. I'll update the earlier sentence to say something like "shows some of the features of columbo".

I'll also add a list of features before jumping into the examples.